### PR TITLE
Helx 11

### PR DESCRIPTION
**Pull request by Mborsellino**
Architecture documentation located at https://renci.atlassian.net/l/cp/sh9zFxyU

### CODE REVIEW

File: tycho/template/pod.yaml [hunk 1]
```diff
@@ -68,6 +68,21 @@ spec:
     volumeMounts:
     - name: {{ system.stdnfs_pvc }}
       mountPath: {{ system.parent_dir }}
+  {% if system.irods_enabled == True %}
+  - name: nfsrods-prep
+    image: busybox:1.28
+    # Fixed resources
+    command: ['sh', '-c'] 
+    args:
+    - touch /permissions/irodsvalidusers &&
```
Comment (pj-linebaugh) [inline tycho/template/pod.yaml:77]: Why is this needed?

File: tycho/template/pod.yaml [hunk 2]
```diff
@@ -68,6 +68,21 @@ spec:
...
+      echo {{system.nfsrods_uid}} >> /permissions/irodsvalidusers
```
Comment (pj-linebaugh) [inline tycho/template/pod.yaml:78]: This might get big after a while.  Maybe check to see if the username is already in there and if it is then don't add it?

File: tycho/template/pod.yaml (hunk 1 above)
Comment (Mborsellino) [reply tycho/template/pod.yaml:77]: This is how I chose to solve the problem that the permissions file needs to be written at a higher privilege level than the user has access to so they can't change the UID. The initContainer runs this script as root.

Comment (pj-linebaugh) [reply tycho/template/pod.yaml:77]: Right after this you echo a line to the same file, which would create it if it didn't exist.  The permissions wouldn't change either way.  Am I missing something?

Comment (Mborsellino) [reply tycho/template/pod.yaml:77]: Ah! I see what you mean. I'll remove that.

File: tycho/template/pod.yaml (hunk 2 above)
Comment (pj-linebaugh) [reply tycho/template/pod.yaml]: Does this file go away when the pod is terminated?  If not, then will it end up having multiple entries for a single username if they start multiple apps?

Comment (Mborsellino) [reply tycho/template/pod.yaml]: Yes, it is in ephemeral storage tied to the pod so it disappears when the pod dies. So with multiple apps started, each pod will get its own permissions file in its own separate PVC. Now, I guess what this brings up that I didn't think about, is how a deployment of NFSRODS will track the creation of these new PVC's.
image

**Comment by waTeim** [comment]
Hye what's a way to turn this on if we want to try to use this feature?

**Review by waTeim — APPROVED** [review]

**Review by pj-linebaugh — APPROVED** [review]
